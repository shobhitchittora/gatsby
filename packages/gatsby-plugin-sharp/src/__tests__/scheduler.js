const { setJobToProcess } = require(`../scheduler`)

describe(`setJobToProcess`, () => {
  it(`allows brackets in paths`, () => {
    let deferred = {}
    deferred.promise = new Promise((resolve, reject) => {
      deferred.resolve = resolve
      deferred.reject = reject
    })
    const toProcess = {}
    const job = {
      args: {},
      inputPath: `1234/file.jpg`,
      contentDigest: `8675309jenny`,
      outputPath: `myoutputpath/1234/file[new].jpg`,
    }
    setJobToProcess(toProcess, job, deferred)
    expect(toProcess).toMatchSnapshot()
  })
})
