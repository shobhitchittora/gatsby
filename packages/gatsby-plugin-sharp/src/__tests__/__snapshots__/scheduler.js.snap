// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setJobToProcess allows brackets in paths 1`] = `
Object {
  "1234/file%2Ejpg": Object {
    "myoutputpath/1234/file[new]%2Ejpg": Object {
      "deferred": Object {
        "promise": Promise {},
        "reject": [Function],
        "resolve": [Function],
      },
      "job": Object {
        "args": Object {},
        "contentDigest": "8675309jenny",
        "inputPath": "1234/file.jpg",
        "outputPath": "myoutputpath/1234/file[new].jpg",
      },
    },
  },
}
`;
